From c55eba802bb4af0b8d9e3d3166dcac78b07b9ec8 Mon Sep 17 00:00:00 2001
From: dragonGR <alex@dragongr.dev>
Date: Wed, 20 Jul 2022 15:35:00 +0300
Subject: [PATCH] NuMediaExtractor: Fix a divide by zero build warning

getTotalBitrate fills in bitrate, however, build warning says something about the possibility being divided by 0.

No way bitrate will get assigned to 0, absolutely no way. So always assume it will be a greater number than 0.

Test: Relevant build warning doesn't show up

Change-Id: Ib079c93e599fce5a35330035b16cbd1f0f5b4cbc
Signed-off-by: Alex Tsanis <alex@dragongr.dev>
---
 media/libstagefright/NuMediaExtractor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libstagefright/NuMediaExtractor.cpp b/media/libstagefright/NuMediaExtractor.cpp
index 68933242a82..814e7583f14 100644
--- a/media/libstagefright/NuMediaExtractor.cpp
+++ b/media/libstagefright/NuMediaExtractor.cpp
@@ -857,7 +857,7 @@ bool NuMediaExtractor::getCachedDuration(
 
     int64_t bitrate;
     if (cachedDataRemaining >= 0
-            && getTotalBitrate(&bitrate)) {
+            && getTotalBitrate(&bitrate) && bitrate > 0) {
         *durationUs = cachedDataRemaining * 8000000ll / bitrate;
         *eos = (finalStatus != OK);
         return true;
